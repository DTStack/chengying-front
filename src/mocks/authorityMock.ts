const authority = [
    'menu_cluster_manage',
    'sub_menu_cluster_manage',
    'cluster_view',
    'cluster_edit',
    'sub_menu_cluster_overview',
    'sub_menu_cluster_host',
    'sub_menu_cluster_image_store',
    'image_store_view',
    'image_store_edit',
    'menu_app_manage',
    'sub_menu_package_manage',
    'package_view',
    'package_upload_delete',
    'sub_menu_installed_app_manage',
    'installed_app_view',
    'menu_deploy_guide',
    'menu_product_overview',
    'menu_service',
    'service_view',
    'service_product_start_stop',
    'service_start_stop',
    'service_roll_restart',
    'service_config_edit',
    'service_config_distribute',
    'service_dashboard_view',
    'menu_product_host',
    'menu_product_diagnosis',
    'sub_menu_log_view',
    'log_view',
    'log_download',
    'sub_menu_event_diagnosis',
    'sub_menu_config_change',
    'menu_monitor',
    'sub_menu_dashboard',
    'sub_menu_alarm',
    'sub_menu_alarm_record',
    'alarm_record_view',
    'alarm_record_open_close',
    'sub_menu_alarm_channel',
    'alarm_channel_view',
    'alarm_channel_edit',
    'menu_user_manage',
    'sub_menu_user_manage',
    'user_view',
    'user_add',
    'user_edit',
    'user_delete',
    'user_able_disable',
    'user_reset_password',
    'sub_menu_role_manage',
    'sub_menu_user_info',
    'menu_security_audit'
];

function getAuthorityList() {
    const obj = {};
    authority.forEach((item: string) => {
        obj[item] = true;
    })
    return obj;
}

export const authorityList = getAuthorityList();
