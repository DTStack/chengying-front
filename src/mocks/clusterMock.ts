export const cur_parent_cluster = {
    id: 46,
    type: 'kubernetes',
    name: 'DTLogger',
    mode: 0
}
export default {

}
