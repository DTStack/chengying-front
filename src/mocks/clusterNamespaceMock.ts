export default {
    getNamespaceList: {
        msg: 'ok',
        code: 0,
        data: [
            {
                id: 26,
                namespace: 'dt-test',
                status: 'valid',
                cpu_used: '1.5core',
                cpu_total: '300core',
                cpu_percent: 0.5,
                memory_used: '2.48GB',
                memory_total: '300GB',
                mem_percent: 0.8294270833333334,
                user: 'admin@dtstack.com',
                update_time: '2020-12-18 17:43:59'
            }
        ]
    }
}
