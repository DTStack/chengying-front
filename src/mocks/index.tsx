export { default as clusterMnagerMock } from './clusterMnagerMock';
export { default as ServiceMock } from './serviceMock';
export { default as ImageStoreMock, imageStoreInfo } from './imageStoreMock';
export { default as ServicePageMock } from './servicePageMock';
export { default as DeployMock } from './deployMock';
export { default as DashBoardMock } from './dashboardMock';
export { default as UserCenterMock } from './userCenterMock';
export { default as SecurityAuditMock } from './securityAuditMock';
export { default as ClusterNamespaceMock } from './clusterNamespaceMock';
export { cur_parent_cluster } from './clusterMock';
export { authorityList } from './authorityMock';
