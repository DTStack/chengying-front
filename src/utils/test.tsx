import * as React from 'react';
import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunkMiddleware from 'redux-thunk';

// this is a handy function that I normally make available for all my tests
// that deal with connected components.
// you can provide initialState for the entire store that the ui is rendered with
export function renderWithRedux(
  Component,
  rootReducer,
  initialState: any = {}
) {
  const store: any = createStore(
    rootReducer,
    initialState,
    applyMiddleware(thunkMiddleware)
  );
  const layout = <Provider store={store}>{Component}</Provider>;
  return {
    ...render(layout),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}
