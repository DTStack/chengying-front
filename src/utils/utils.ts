import { message } from 'antd';
import * as Cookies from 'js-cookie';
import { COOKIES } from '@/constants/const';

// tslint:disable
const utils = {
  // 公用的方法
  // loginOut: function () {
  //     Http.get(req_url.login_out,{}).then(ret => {
  //         Http.get(req_url.get_config,{}).then(data => {
  //             window.location.href = uicUrl + ":" + uicPort + redirectLink + window.location.href;
  //         })
  //         return ret;
  //     })
  // },
  /**
   * 获取页面宽度
   * @return {[type]} [description]
   */
  pageWidth: function () {
    return Math.max(
      document.documentElement.clientWidth,
      window.innerWidth || 0
    );
  },

  /**
   * 获取页面高度
   * @return {[type]} [description]
   */
  pageHeight: function () {
    return Math.max(
      document.documentElement.clientHeight,
      window.innerHeight || 0
    );
  },

  /**
   * 根据参数名获取URL数据
   * @param  {[type]} name [description]
   * @param  {[type]} url  [description]
   * @return {[type]}      [description]
   */
  getParameterByName: function (name: any, url: any) {
    if (!url) {
      url = window.location.href;
    }
    name = name.replace(/[[\]]/g, '\\$&');
    var regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)');
    var results = regex.exec(url);
    if (!results) {
      return null;
    }
    if (!results[2]) {
      return '';
    }
    return decodeURIComponent(results[2].replace(/\+/g, ' '));
  },

  /**
   * 获取图片的Base64格式
   * @param  {[type]}   img      [description]
   * @param  {Function} callback [description]
   * @return {[type]}            [description]
   */
  getBase64: function (img: any, callback: any) {
    const reader = new FileReader();
    reader.addEventListener('load', () => callback(reader.result));
    reader.readAsDataURL(img);
  },

  /**
   * 百分比转换
   * @param  {[type]} num       [description]
   * @param  {[type]} precision [description]
   * @return {[type]}           [description]
   */
  percent: function (num: any, precision: any) {
    if (!num || num === Infinity) {
      return 0 + '%';
    }
    if (num > 1) {
      num = 1;
    }
    precision = precision || 2;
    precision = Math.pow(10, precision);
    return Math.round(num * precision * 100) / precision + '%';
  },

  getCssText: function (object: any) {
    var str = '';
    for (var attr in object) {
      str += attr + ':' + object[attr] + ';';
    }
    return str;
  },

  formateDateTime: function (timestap: any) {
    if (!timestap) {
      return '';
    }
    const times = new Date(timestap);
    const year = times.getFullYear();
    let month = times.getMonth() + 1 + '';
    let day = times.getDate() + '';

    let hour = times.getHours() + '';
    let minutes = times.getMinutes() + '';
    let seconds = times.getSeconds() + '';

    if (month.toString().length === 1) {
      month = `0${month}`;
    }

    if (day.toString().length === 1) {
      day = `0${day}`;
    }

    if (hour.toString().length === 1) {
      hour = `0${hour}`;
    }

    if (minutes.toString().length === 1) {
      minutes = `0${minutes}`;
    }

    if (seconds.toString().length === 1) {
      seconds = `0${seconds}`;
    }
    return `${year}-${month}-${day} ${hour}:${minutes}:${seconds}`;
  },

  formateDate: function (timestap: any) {
    if (!timestap) {
      return '';
    }
    const times = new Date(timestap);
    const year = times.getFullYear();
    let month = times.getMonth() + 1 + '';
    let day = times.getDate() + '';
    if (month.toString().length === 1) {
      month = `0${month}`;
    }
    if (day.toString().length === 1) {
      day = `0${day}`;
    }
    return `${year}-${month}-${day}`;
  },

  trim: function (str: any) {
    return typeof str === 'string'
      ? str.replace(/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g, '')
      : str;
  },

  messageFilter: function (
    data: any,
    callback: any,
    info?: any,
    hideSuccessMessage?: any
  ) {
    if (data.status.code === 1001) {
      // if(info){
      //     message.success(info);
      // }
      // message.success(data.status.msg);
      if (!hideSuccessMessage) {
        message.success('成功');
      }
      if (callback) {
        callback();
      }
    } else {
      // console.log(data.status.msg)
      // message.error(data.status.msg);
    }
  },

  // filter查询条件的构造
  getTableParam: (filter: any) => {
    const param = {};
    filter.forEach((item: any, i: any) => {
      const keyArr = item.split('-');
      param[keyArr[1]] = keyArr[2];
    });
    return param;
  },

  // 抑制告警
  setAlertControl: (data: any) => {
    const alarmSuppression: any = {};
    const acKey = data.alert_control;
    if (acKey === '1') {
      alarmSuppression.close = 1;
    } else {
      alarmSuppression.close = 0;
      if (data['alert_num_' + acKey]) {
        alarmSuppression.times_after_close = data['alert_num_' + acKey];
      }
      if (data['recover_num_' + acKey]) {
        alarmSuppression.close_after_long_time_reset_amount =
          data['recover_num_' + acKey];
        alarmSuppression.close_after_long_time_reset_unit =
          data['recover_unit_' + acKey];
      }
    }
    return alarmSuppression;
  },
  getAlertControl: (alertControl: any) => {
    const fields = {
      alert_control: '',
    };
    if (alertControl.close === 0) {
      if (alertControl.times_after_close !== 0) {
        fields.alert_control = '2';
        fields['alert_num_' + fields.alert_control] =
          alertControl.times_after_close;
      } else {
        fields.alert_control = '3';
      }
      if (alertControl.close_after_long_time_reset_amount !== '') {
        if (fields.alert_control === '2') {
          fields.alert_control = '4';
        }
        fields['alert_num_' + fields.alert_control] =
          alertControl.times_after_close;
        fields['recover_num_' + fields.alert_control] =
          alertControl.close_after_long_time_reset_amount;
        fields['recover_unit_' + fields.alert_control] =
          alertControl.close_after_long_time_reset_unit;
      }
    }
    return fields;
  },

  // 模糊匹配
  filterOption: (inputValue: any, option: any) => {
    if (option.props.children.indexOf(inputValue) !== -1) {
      return true;
    } else {
      return false;
    }
  },
  // 从url中分离参数
  getParamsFromUrl(url: any) {
    var obj = {};
    var keyvalue = [];
    var key = '';
    var value = '';
    var paraString = url.substring(url.indexOf('?') + 1, url.length).split('&');
    for (var i in paraString) {
      keyvalue = paraString[i].split('=');
      key = keyvalue[0];
      value = keyvalue[1];
      obj[key] = value;
    }
    return obj;
  },
  jsonToQuery(obj: any) {
    function cleanArray(actual: any) {
      const newArray = [];
      for (let i = 0; i < actual.length; i++) {
        if (actual[i]) {
          newArray.push(actual[i]);
        }
      }
      return newArray;
    }
    if (!obj) {
      return '';
    }
    return cleanArray(
      Object.keys(obj).map((key) => {
        if (obj[key] === undefined) {
          return '';
        }
        return encodeURIComponent(key) + '=' + encodeURIComponent(obj[key]);
      })
    ).join('&');
  },
  // 判断Object是否为{}
  checkNullObj(obj: any) {
    return Object.keys(obj).length === 0;
  },

  // service中对于简单res的处理
  serviceCallback(res: any, successMsg?: string, callback?: Function) {
    if (res.data.code === 0) {
      message.success(successMsg);
      callback && callback();
    } else {
      message.error(res.data.msg);
    }
  },

  // 是否有权限操作
  noAuthorityToDO(authorityList: any, type: string) {
    if (!authorityList[type]) {
      message.error('权限不足，请联系管理员!');
      return true;
    }
    return false;
  },

  get k8sNamespace(): string {
    return Cookies.get(COOKIES.NAMESPACE);
  },

  setNaviKey: (top: string | undefined, sub: string | undefined): void => {
    sessionStorage.setItem('firstLevelNav', top);
    sessionStorage.setItem('siderLevelNav', sub);
  },

  // 转化为GB
  formatGBUnit: (r: string): number => {
    let result: number = 0;
    if (r.indexOf('MB') > -1) {
      result = parseFloat(r.replace('MB', '')) / 1024;
    } else if (r.indexOf('KB') > -1) {
      result = parseFloat(r.replace('KB', '')) / 1024 / 1024;
    } else if (r.indexOf('GB') > -1) {
      result = parseFloat(r.replace('GB', ''));
    } else if (r.indexOf('TB') > -1) {
      result = parseFloat(r.replace('TB', '')) * 1024;
    } else {
      result = 0;
    }
    return result;
  },
};

export default utils;
