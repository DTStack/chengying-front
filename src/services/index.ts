import apis from '@/constants/apis';
import * as http from '@/utils/http';

const { product, service, instance, host, alert, cluster, common } = apis;

export const Service = {
  addNetworkPackage(params: any) {
    return http[product.addNetworkPackage.method](
      product.addNetworkPackage.url,
      params
    );
  },
  getProductpackageList() {
    return http[product.getProductpackageList.method](
      product.getProductpackageList.url
    );
  },
  delProductPackageItem(params: any) {
    return http[product.delProductpackageItem.method](
      product.delProductpackageItem.url,
      params
    );
  },
  getAllProducts(params: any) {
    return http[product.getProductList.method](
      product.getProductList.url,
      params
    );
  },
  Verifylink(params: any) {
    return http[product.Verifylink.method](product.Verifylink.url, params);
  },
  // 获取更新目录
  getPatchPath(params: any) {
    return http[product.getPatchPath.method](product.getPatchPath.url, params);
  },
  // 上传更新接口
  updatePatchPath(params: any) {
    return http[product.updatePatchPath.method](
      product.updatePatchPath.url,
      params
    );
  },
  // 获取更新状态
  updatePatchStatus(params: any) {
    return http[product.updatePatchStatus.method](
      product.updatePatchStatus.url,
      params
    );
  },
  getParentProductList(params?: any) {
    return http[product.getParentProductList.method](
      product.getParentProductList.url,
      params
    );
  },
  // 获取产品列表
  getClusterProductList(params?: any) {
    return http[product.getClusterProductList.method](
      product.getClusterProductList.url,
      params
    );
  },
  getCurrentProduct(params: any) {
    return http[product.getCurrentProduct(params).method](
      product.getCurrentProduct(params).url,
      params
    );
  },
  getProductUpdateRecords(params: any) {
    return http[product.getProductUpdateRecords(params).method](
      product.getProductUpdateRecords(params).url,
      params
    );
  },
  // 获取补丁包更新历史
  getProductUpdate(params: any) {
    return http[product.getProductUpdate(params).method](
      product.getProductUpdate(params).url,
      params
    );
  },
  // 获取产品包升级目标版本列表
  getProductVersionList(params: any) {
    const { upgrade_mode } = params;
    return http[product.getProductVersionList(params).method](
      product.getProductVersionList(params).url,
      {
        upgrade_mode,
      }
    );
  },
  getProductDeployRecords(params: any) {
    return http[product.getProductDeployRecords(params).method](
      product.getProductDeployRecords(params).url,
      params
    );
  },
  getServiceDashInfo(params: any) {
    const url = service.getServiceDashInfo.url;
    const { product_name, service_name } = params;
    return http[service.getServiceDashInfo.method](
      `${url}?tag=${product_name}&tag=${service_name}`
    );
  },
  getInstanceLog(params: any) {
    return http[instance.getInstanceLog(params).method](
      instance.getInstanceLog(params).url,
      params
    );
  },
  getFilterData(params: any) {
    return http[host.getFilterData.method](host.getFilterData.url, params);
  },
  // 获取主机集群下的主机列表
  getHostClusterHostList(params: any) {
    return http[common.getHostClusterHostList.method](
      common.getHostClusterHostList.url,
      params
    );
  },
  // 获取kubernetes集群下的主机列表
  getKubernetesClusterHostList(params: any) {
    return http[common.getKubernetesClusterHostList.method](
      common.getKubernetesClusterHostList.url,
      params
    );
  },
  // 获取主机列表
  getClusterHostList(params: any, type: string) {
    return type === 'hosts'
      ? this.getHostClusterHostList(params)
      : this.getKubernetesClusterHostList(params);
  },
  // 获取集群下主机分组列表
  getClusterhostGroupLists(params: any) {
    return http[common.getClusterhostGroupLists.method](
      common.getClusterhostGroupLists.url,
      params
    );
  },
  getTableData(params: any) {
    return http[host.getTableData.method](host.getTableData.url, params);
  },
  showInstallInfo(params: any) {
    return http[host.showInstallInfo.method](host.showInstallInfo.url, params);
  },
  getAlertsHistory(params: any) {
    return http[alert.getAlertsHistory(params).method](
      alert.getAlertsHistory(params).url,
      params
    );
  },
  getHostInstance(params: any) {
    return http[host.getHostInstance.method](host.getHostInstance.url, params);
  },
  getHostServicesList(params: any) {
    return http[host.getHostServicesList.method](
      host.getHostServicesList.url,
      params
    );
  },
  getDeployShot(params: any) {
    return http[product.getDeployShot(params).method](
      product.getDeployShot(params).url
    );
  },
  getPatchUpdateShot(params: any) {
    return http[product.getPatchUpdateShot(params).method](
      product.getPatchUpdateShot(params).url
    );
  },
  delProduct(params: any) {
    return http[product.delProduct(params).method](
      product.delProduct(params).url
    );
  },
  getClusterList(params: any) {
    return http[cluster.getClusterList.method](
      cluster.getClusterList.url,
      params
    );
  },
};

export { default as productService } from './productService';
export { default as dashboardService } from './dashboardService';
export { default as alertRuleService } from './alertRuleService';
export { default as alertChannelService } from './alertChannelService';
export { default as installGuideService } from './installGuideService';
export { default as addHostService } from './addHostService';
export { default as deployService } from './deployService';
export { default as servicePageService } from './ServicePageService';
export { default as createHostService } from './createHostService';
export { default as userCenterService } from './userCenterService';
export { default as eventService } from './eventService';
export { default as configService } from './configService';
export { default as clusterManagerService } from './clusterManagerService';
export { default as clusterHostService } from './clusterHostService';
export { default as imageStoreService } from './imageStoreService';
export { default as SecurityAuditService } from './securityAuditService';
export { default as ClusterNamespaceService } from './clusterNamespaceService';
export { default as inspectionReportService } from './inspectionReportService';
export { default as echoService } from './echoService';
export { default as securityService } from './securityService';
export { default as scriptManager } from './scriptManager';
export { default as globalConfig } from './globalConfig';
export { default as productLine } from './productLine';