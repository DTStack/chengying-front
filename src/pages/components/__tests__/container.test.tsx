import * as React from 'react';
import Container from '../container';
import { renderWithRedux } from '@/utils/test';
import reducer from '@/stores';
import { Service } from '@/services';
import { ServiceMock } from '@/mocks';
import { fireEvent, RenderResult, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

jest.mock('@/services');

describe('app deploy list', () => {
  let wrapper: RenderResult;
  // 初始数据
  const clusterList = ServiceMock.getClusterProductList.data;
  const firstCluster = clusterList[0];
  // const secondCluster = clusterList[1];
  beforeEach(() => {
    (Service.getClusterProductList as any).mockResolvedValue({
      data: ServiceMock.getClusterProductList,
    });
    (Service.getAllProducts as any).mockResolvedValue({
      data: ServiceMock.getAllProducts,
    });
    wrapper = renderWithRedux(<Container />, reducer, {});
  });

  afterEach(cleanup);

  test('const Snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  // 下拉筛选
  test('select unit test', () => {
    const selects = wrapper.container.getElementsByClassName(
      'ant-select-selection__rendered'
    );
    // render
    for (const i in selects) {
      const list = [
        {
          placeholder: '选择集群',
          value: '选择集群', // DTLogger
        },
        {
          placeholder: '选择产品',
          value: firstCluster.subdomain.products[0], // DTEM
        },
        {
          placeholder: '选择组件',
          value: '',
        },
      ];
      if (selects[i] instanceof HTMLElement) {
        expect(selects[i].firstChild.textContent).toBe(list[i].placeholder);
        list[i].value &&
          expect(selects[i].lastChild.textContent).toBe(list[i].value);
      }
    }
    /* 下拉选择 - 以集群为例 */
    // const clusterSelect = wrapper.container.getElementsByClassName('ant-select')[0];
    // fireEvent.click(clusterSelect);
    // 下拉菜单
    // const clusterOptions = wrapper.getByTestId(`cluster-option-${secondCluster.clusterName}`);
    // expect(clusterOptions).toBeInTheDocument();
    // 选择
    // fireEvent.click(clusterOptions);
    // expect(selects[0].lastChild.textContent).toBe(secondCluster.clusterName);
    expect(Service.getAllProducts).toHaveBeenCalled();
    // expect(Service.getAllProducts).toHaveBeenLastCalledWith({
    //     clusterId: secondCluster.clusterId,
    //     parentProductName: secondCluster.subdomain.DTinsight[0],
    //     limit: 0,
    //     mode: 1,
    //     productName: undefined,
    //     productVersion: undefined,
    //     componentList: []
    // });
  });

  // 版本号搜索
  test('search input', () => {
    const searchInput = wrapper.getByPlaceholderText('按组件版本号搜索');
    const icon = searchInput.nextSibling.firstChild;
    expect(searchInput).toBeInTheDocument();
    // 搜索
    fireEvent.change(searchInput, { target: { value: '_beta' } });
    fireEvent.click(icon);
    expect(Service.getAllProducts).toHaveBeenCalled();
  });
});
