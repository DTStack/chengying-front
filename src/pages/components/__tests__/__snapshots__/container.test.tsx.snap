// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app deploy list const Snapshot 1`] = `
<DocumentFragment>
  <div
    class="content-container"
  >
    <div
      class="mb-12 clearfix"
    >
      <div
        style="margin-bottom: 10px; display: inline-block;"
      >
        <span
          class="mr-20"
        >
          集群：
          <div
            class="dt-form-shadow-bg ant-select ant-select-disabled"
            style="width: 180px;"
          >
            <div
              aria-autocomplete="list"
              aria-controls="test-uuid"
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-select-selection
            ant-select-selection--single"
              role="combobox"
              tabindex="-1"
            >
              <div
                class="ant-select-selection__rendered"
              >
                <div
                  class="ant-select-selection__placeholder"
                  style="display: none; user-select: none;"
                  unselectable="on"
                >
                  选择集群
                </div>
                <div
                  class="ant-select-selection-selected-value"
                  style="display: block; opacity: 1;"
                  title="选择集群"
                >
                  选择集群
                </div>
              </div>
              <span
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <i
                  aria-label="icon: down"
                  class="anticon anticon-down ant-select-arrow-icon"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </i>
              </span>
            </div>
          </div>
        </span>
        <span
          class="mr-20"
        >
          产品：
          <div
            class="dt-form-shadow-bg ant-select ant-select-enabled"
            style="width: 180px; font-size: 12px;"
          >
            <div
              aria-autocomplete="list"
              aria-controls="test-uuid"
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-select-selection
            ant-select-selection--single"
              role="combobox"
              tabindex="0"
            >
              <div
                class="ant-select-selection__rendered"
              >
                <div
                  class="ant-select-selection__placeholder"
                  style="display: none; user-select: none;"
                  unselectable="on"
                >
                  选择产品
                </div>
                <div
                  class="ant-select-selection-selected-value"
                  style="display: block; opacity: 1;"
                  title=""
                >
                  <i
                    aria-label="icon: appstore-o"
                    class="anticon anticon-appstore-o"
                    style="margin-right: 6px;"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="appstore"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M464 144H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H212V212h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H612V212h200v200zM464 544H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H212V612h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H612V612h200v200z"
                      />
                    </svg>
                  </i>
                  DTEM
                </div>
              </div>
              <span
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <i
                  aria-label="icon: down"
                  class="anticon anticon-down ant-select-arrow-icon"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </i>
              </span>
            </div>
          </div>
        </span>
        <span
          class="mr-20"
        >
          组件名称：
          <div
            class="dt-form-shadow-bg ant-select ant-select-enabled"
            style="width: 180px;"
          >
            <div
              aria-autocomplete="list"
              aria-controls="test-uuid"
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-select-selection
            ant-select-selection--multiple"
              role="combobox"
            >
              <div
                class="ant-select-selection__rendered"
              >
                <div
                  class="ant-select-selection__placeholder"
                  style="display: block; user-select: none;"
                  unselectable="on"
                >
                  选择组件
                </div>
                <ul>
                  <li
                    class="ant-select-search ant-select-search--inline"
                  >
                    <div
                      class="ant-select-search__field__wrap"
                    >
                      <input
                        autocomplete="off"
                        class="ant-select-search__field"
                        value=""
                      />
                      <span
                        class="ant-select-search__field__mirror"
                      >
                         
                      </span>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </span>
        <span
          class="mr-20"
        >
          <span
            class="ant-input-search dt-form-shadow-bg ant-input-affix-wrapper"
            style="width: 264px;"
          >
            <input
              class="ant-input"
              placeholder="按组件版本号搜索"
              type="text"
              value=""
            />
            <span
              class="ant-input-suffix"
            >
              <i
                aria-label="icon: search"
                class="anticon anticon-search ant-input-search-icon"
                tabindex="-1"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="search"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                  />
                </svg>
              </i>
            </span>
          </span>
        </span>
      </div>
      <button
        class="ant-btn fl-r ant-btn-primary"
        style="margin-left: 20px;"
        type="button"
      >
        <span>
          上传补丁包
        </span>
      </button>
      <button
        class="ant-btn fl-r ant-btn-primary"
        type="button"
      >
        <i
          aria-label="icon: download"
          class="anticon anticon-download"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="download"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M505.7 661a8 8 0 0 0 12.6 0l112-141.7c4.1-5.2.4-12.9-6.3-12.9h-74.1V168c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v338.3H400c-6.7 0-10.4 7.7-6.3 12.9l112 141.8zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </i>
        <span>
          下载产品部署内容
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
